Refactor

andriikonoh opened the pull request:
#Refactor

andriikonoh committed:
Move implementations to specific classes

andriikonoh committed:
Implement the ability to use offers before closing a check

andriikonoh committed:
Implement the ability to use offers before closing a check

andriikonoh committed:
Implement expiration date cheker

andriikonoh committed:
Implement StrategyRealisation

andriikonoh committed:
refactor

andriikonoh committed:
refactor

alexanderko commented on src/checkout/Offer.java:
```diff
@@ -1,5 +1,27 @@
+    }
+
+    public Offer(Reward reward , Condition condition){
+        this(reward, condition, LocalDate.of(2019, 3, 7));
```
hard-coded value in production code

alexanderko commented on src/checkout/CheckoutService.java:
```diff
@@ -16,23 +20,26 @@ public void addProduct(Product product) {
-        }
+    private void useAllOffer(){
+        if(this.offers.size()!=0) this.offers.forEach(offer ->{
+            if(isOfferavAilable(offer)) offer.apply(check);
```
spelling

alexanderko commented on src/checkout/CheckoutService.java:
```diff
@@ -16,23 +20,26 @@ public void addProduct(Product product) {
+
+    }
+    public boolean isOfferavAilable(Offer ofer){
+        if (ofer.getExpirationDate().isAfter(check.getTodayDate()))
```
spelling

alexanderko replied:
Redundant if statement

alexanderko commented on src/checkout/Check.java:
```diff
@@ -1,11 +1,15 @@
     private List<Product> products = new ArrayList<>();
     private int points = 0;
+    private int discount = 0;
+    private LocalDate todayDate = LocalDate.of(2019, 3, 1);
```
hard-coded value in production code

alexanderko reviewed (changes requested):
- [ ] fix code formatting
- [ ] fix spelling
- [ ] make test names more descriptive 
- [ ] cleanup production code from hard-coded dates

andriikonoh committed:
refactor(all project and test)
fix code formatting
fix spelling
make test names more descriptive
cleanup production code from hard-coded dates
